File: libs/monitoring/features/change-source-binding-status/change-source-binding-status.command.ts
import type { TenantId, WorkspaceId } from '@social-monitor/shared-kernel';

import type { SourceBindingStatus } from '../../domain';

export type ChangeSourceBindingStatusCommand = {
  readonly tenantId: TenantId;
  readonly workspaceId: WorkspaceId;
  readonly topicId: string;
  readonly sourceBindingId: string;
  readonly status: SourceBindingStatus;
  readonly idempotencyKey: string;
  readonly correlationId: string;
};

File: apps/frontend/features/sources/lib/src/application/commands/change_source_binding_status_command.dart
import 'package:social_monitor_shared_kernel/social_monitor_shared_kernel.dart';

import '../../domain/value_objects/source_binding_id.dart';
import '../../domain/value_objects/source_binding_status.dart';
import '../../domain/value_objects/source_interest_id.dart';

final class ChangeSourceBindingStatusCommand {
  const ChangeSourceBindingStatusCommand({
    required this.scope,
    required this.interestId,
    required this.sourceBindingId,
    required this.status,
    required this.idempotencyKey,
  });

  final WorkspaceScope scope;
  final SourceInterestId interestId;
  final SourceBindingId sourceBindingId;
  final SourceBindingStatus status;
  final String idempotencyKey;
}
